From cc5f1d0759b367265a1a000287e2ec15c31eb518 Mon Sep 17 00:00:00 2001
From: Mingli Yu <mingli.yu@windriver.com>
Date: Mon, 26 Feb 2024 14:56:02 +0800
Subject: [PATCH] Remove the compile_time_assert lines

Remove the problematic compile_time_assert lines to fix the below build
failure on 32-bit arm.
 In file included from TOPDIR/build/tmp/work/cortexa15t2hf-neon-yoe-linux-gnueabi/mariadb/10.11.7/mariadb-10.11.7/tests/mysql_client_test.c:38:
 TOPDIR/build/tmp/work/cortexa15t2hf-neon-yoe-linux-gnueabi/mariadb/10.11.7/mariadb-10.11.7/tests/mysql_client_fw.c:1438:3: error: 'compile_time_assert' declared as an array with a negative size
  1438 |   compile_time_assert(sizeof(MYSQL) == 77*sizeof(void*)+656);
       |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Inappropriate [Upstream will bring the asset back
                                in a new way [1]]
[1] https://jira.mariadb.org/browse/MDEV-33429

Signed-off-by: Mingli Yu <mingli.yu@windriver.com>
---
 tests/mysql_client_fw.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/tests/mysql_client_fw.c b/tests/mysql_client_fw.c
index c9e64678..5c0c7ce2 100644
--- a/tests/mysql_client_fw.c
+++ b/tests/mysql_client_fw.c
@@ -1430,14 +1430,6 @@ int main(int argc, char **argv)
     tests_to_run[i]= NULL;
   }
 
-#ifdef _WIN32
-  /* must be the same in C/C and embedded, 1208 on 64bit, 968 on 32bit */
-  compile_time_assert(sizeof(MYSQL) == 60*sizeof(void*)+728);
-#else
-  /* must be the same in C/C and embedded, 1272 on 64bit, 964 on 32bit */
-  compile_time_assert(sizeof(MYSQL) == 77*sizeof(void*)+656);
-#endif
-
   if (mysql_server_init(embedded_server_arg_count,
                         embedded_server_args,
                         (char**) embedded_server_groups))
-- 
2.25.1

